proxy.py

```diff
@@ -22,20 +28,31 @@
         app.whitelist = set(line.rstrip() for line in f)
     app.forward = args.forward
 
+    formatter = logging.Formatter('[%(asctime)s] %(levelname)s in %(module)s: %(message)s')
+    handler = RotatingFileHandler(args.log, maxBytes=0x100000, backupCount=3)
+    handler.setLevel(logging.INFO)
+    handler.setFormatter(formatter)
+    app.logger.addHandler(handler)
+    app.logger.setLevel(logging.INFO)
+
     @app.route('/', methods = ['POST'])
     def proxy():
-        if not request.is_json: # Invalid JSON Format
-            abort(404)
+        if not request.is_json: # Content type is not json 
+            app.logger.info('Received the data which is not JSON [{}]'.format(request.remote_addr))
+            abort(400)
```

Comment:
Why do you change 404 Not Found to 400 Bad Request?

Reply:
Read the commit message. 2d1d172

Reply:
Divided into two commits.